Automattic/wp-calypso
# Domains: Create name servers component in domain settings page

### Changes proposed in this Pull Request

This PR creates the name servers card in the main domain settings page. It is basically an updated, functional adaptation of the `domains/domain-management/name-servers` and `domains/domain-management/name-servers/wpcom-nameservers-toggle` components with some improvements.

This is part of the domain settings pages redesign project detailed in pcYYhz-m2-p2.

#### Screenshots

- New section:

![Markup on 2021-12-20 at 23:52:30](https://user-images.githubusercontent.com/5324818/146864825-0d40ffc8-5069-47e7-91d1-75e46fbba515.png)

- New section with custom name servers form open:

<img width="715" alt="Screen Shot 2021-12-20 at 23 53 47" src="https://user-images.githubusercontent.com/5324818/146864846-387bb5ee-ded5-40bc-ab38-d2d0faecf154.png">

<img width="704" alt="Screen Shot 2021-12-20 at 23 55 53" src="https://user-images.githubusercontent.com/5324818/146864879-77b10748-37fe-42e2-b8b0-7ecc2e0fee85.png">

- New section with custom name servers saved:

<img width="706" alt="Screen Shot 2021-12-20 at 23 54 13" src="https://user-images.githubusercontent.com/5324818/146864907-2aa84603-e3b6-46da-9a14-942e825a5100.png">

- Old page:

![Markup on 2021-12-21 at 00:07:25](https://user-images.githubusercontent.com/5324818/146864924-a8b46e70-b4a7-4014-bb67-a89a3b822f44.png)

- Old page with custom name servers form open:

<img width="1019" alt="Screen Shot 2021-12-21 at 00 07 58" src="https://user-images.githubusercontent.com/5324818/146864949-0879504f-b4c9-4d07-a5cf-4e8aff0062c3.png">

### Testing instructions

- Build this branch locally or open the live Calypso link
  - If you're in the live Calypso link, please append `?flags=domains/settings-page-redesign` to your URL to enable the appropriate feature flag
- Go to "Upgrades > Domains"
- Select one of your domains

All the original functionalities should be working the same, so please try to:
- Set custom name servers
  - Add up to 4 name servers
- Reset to WPCOM name servers after setting custom name servers by enabling the "Point to WPCOM" toggle again
- Start changing name servers and then cancel editing
- Save invalid name servers (empty list or invalid domains) and ensure they are not saved
- Check different site types (simple, atomic, domain-only) and domain types (registered, mapped, transfer) - only **registered** domains should show the name servers card
- Also ensure that the mobile view looks correct
